scripts/reborn_webui_v2_live_qa/test_run_live_qa.py

```
@@ -3505,6 +3505,626 @@ def test_generated_slack_home_uses_webui_setup_only(self):
+            ),
+            "xoxp-second-identity",
+        )
+        empty_env = {
+            "AUTH_LIVE_SLACK_SECOND_USER_TOKEN": "",
+            "AUTH_LIVE_SLACK_SECOND_USER_TOKEN_PATH": "",
+        }
+        with patch.dict(os.environ, empty_env, clear=False):
+            self.assertIsNone(run_live_qa._slack_second_user_token({}))
+            # Arms that strictly need a second HUMAN must fail with this exact
+            # precondition message — never silently skip.
+            with self.assertRaisesRegex(
```

Ruff (0.15.20)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)